src/utils/extensionParsing.ts

```diff
@@ -554,10 +619,33 @@ export function getEntityMetricPatterns(typeIdx: number, extension: ExtensionStu
   return patterns;
 }
 
+/**
+ * Extracts all required dimensions for Metrics sourceType across all rules of a given
+ * topology type. The topology type must be referenced by index.
+ * @param typeIdx index of the topology type
+ * @param extension extension.yaml serialized as object
+ * @returns list of required dimensions
+ */
+export function getEntityRequiredDimensions(typeIdx: number, extension: ExtensionStub) {
```

Comment:
could the existing `getRequiredDimensions` not have been used instead of writing another function?